src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/ClientConfigPaths.cs

```diff
@@ -48,7 +49,14 @@ private ClientConfigPaths(string exePath, bool includeUserConfig)
                 // Exe path wasn't specified, get it from the entry assembly
                 exeAssembly = Assembly.GetEntryAssembly();
 
-                if (exeAssembly != null)
+                // in case of SingleFile deployment, Assembly.Location is empty.
+                if (exeAssembly != null && string.IsNullOrEmpty(exeAssembly.Location))
```

Comment:
```C#
if (exeAssembly?.Location.Length > 0)
{
    HasEntryAssembly = true;
}
else
{
   ...
```

Reply:
I have simplified this condition as `if (exeAssembly?.Location.Length == 0)`, but the rest of the flow we need because `isSingleFile` is used below.